src/main/java/org/prebid/server/bidder/intertech/IntertechBidder.java

```diff
@@ -0,0 +1,237 @@
+        final List<BidderError> errors = new ArrayList<>();
+
+        final String referer = getReferer(request);
+        final List<String> curs = request.getCur();
+        final String cur = curs != null && !curs.isEmpty() ? curs.get(0) : "";
+
+        final List<Imp> impList = request.getImp();
+        for (int i = 0; i < impList.size(); i++) {
+            try {
+                final Imp imp = impList.get(i);
+                ExtImpIntertech extImpIntertech = parseAndValidateImpExt(imp.getExt(), i);
+                final Imp modifiedImp = modifyImp(imp);
+                final String modifiedUrl = modifyUrl(extImpIntertech, referer, cur);
+                final BidRequest modifiedRequest =
+                        request.toBuilder().imp(Collections.singletonList(modifiedImp)).build();
+                bidRequests.add(buildHttpRequest(modifiedRequest, modifiedUrl));
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+            }
+        }
+        return Result.of(bidRequests, errors);
+    }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(BidderCall<BidRequest> bidderCall, BidRequest bidRequest) {
+        try {
+            final BidResponse bidResponse = mapper.decodeValue(bidderCall.getResponse().getBody(), BidResponse.class);
+            return Result.of(extractBids(bidResponse, bidderCall.getRequest().getPayload()), Collections.emptyList());
+        } catch (DecodeException | PreBidException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private static List<BidderBid> extractBids(BidResponse bidResponse, BidRequest bidRequest) {
+        final List<SeatBid> seatBids = bidResponse != null ? bidResponse.getSeatbid() : null;
+        if (seatBids == null) {
+            return Collections.emptyList();
+        }
+
+        if (seatBids.isEmpty()) {
+            throw new PreBidException("SeatBids is empty");
+        }
+        return bidsFromResponse(bidResponse, bidRequest.getImp());
+    }
+
+    private static List<BidderBid> bidsFromResponse(BidResponse bidResponse, List<Imp> imps) {
+        return bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .map(SeatBid::getBid)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .map(bid -> BidderBid.of(bid, getBidType(bid.getImpid(), imps), bidResponse.getCur()))
+                .collect(Collectors.toList());
+    }
+
+    private static BidType getBidType(String bidImpId, List<Imp> imps) {
+        for (Imp imp : imps) {
+            if (bidImpId.equals(imp.getId())) {
+                final BidType bidType = resolveImpType(imp);
+                if (bidType == null) {
+                    throw new PreBidException("Processing an invalid impression; cannot resolve impression type");
+                }
+                return bidType;
```

Comment:
Can be a simple return statement if you throw exception directly in `resolveImpType`